### DISCUSSION TITLE
DataPointTooltip problem

### DISCUSSION THREAD

**Discussion by sangiorgi-inaf** [GSI-CS-CO/chart-fx#662] [Q&A]
Hi, I've problem with DataPointTooltip in XYChart.
I use this code to enable it:
		final DataPointTooltip dataTip = new DataPointTooltip();
		chart.getPlugins().add(dataTip);
but no tooltip with data appears.

Current configuration of the project is: jdk 17, javafx 18 and chartfx 11.3.

any suggestion?
thanks in advance

**Comment by SVashdy** [comment]
I believe you should add a picking distance as a parameter. This example worked for me: 

`chart.getPlugins().add(new DataPointTooltip(10));`

**Reply by wirew0rm** [reply]
Thanks for the help! In this case from the code the picking distance should be initialized to 5, so unless there is something (bug?) that prevents low picking distances this is probably not the root cause.

**Comment by wirew0rm** [comment]
Hey and welcome to chart-fx!

From the info you provided I cannot see anything wrong, but you are missing the information about which dataset(s) you added to your chart. Since the dataset API is very flexible, there might be some combination that has not been properly tested with the tooltip plugin.

What I like to do in these cases is to start from a sample which uses the particular feature and then adapt that by removing everything not needed by my usecase and changing everything that is different to my application and see where it breaks. Feel free to share this modified sample here to help us reproduce the problem and improve the library.

**Reply by sangiorgi-inaf** [reply]
parameter in the DataPointTooltip doesn't work for me.

Here is my code:

```java
public class ScanPlotterGuiController implements Initializable {
	@FXML
	VBox vbox;

	final DefaultNumericAxis xAxis = new DefaultNumericAxis("Threshold", "pe");
	final DefaultNumericAxis yAxis = new DefaultNumericAxis("Camera Rate", "evt/s");
	XYChart chart = new XYChart(xAxis, yAxis);

	ArrayList<double[]> rateData = new ArrayList<double[]>();

	@Override
	public void initialize(URL location, ResourceBundle resources) {

		chart.getToolBar().setDisable(true);
		chart.getToolBar().setVisible(false);
		
		final DataPointTooltip dataTip = new DataPointTooltip(10);
		chart.getPlugins().add(dataTip);

		final Zoomer zoomer = new Zoomer();
		zoomer.setPannerEnabled(false);
		zoomer.setSliderVisible(false);
		chart.getPlugins().add(zoomer);
		
		chart.getPlugins().add(new EditAxis());

		xAxis.setAutoRanging(true);
//		xAxis.setAutoGrowRanging(true);
		
		yAxis.setLogAxis(true);
		yAxis.setForceZeroInRange(true);
		yAxis.setAutoRanging(true);
//		yAxis.setAutoGrowRanging(true);
		
		vbox.getChildren().add(chart);
		vbox.setVgrow(chart, Priority.ALWAYS);

	}

	public void plotData(ArrayList<double[]> rateData) {
		this.rateData = rateData;

		DoubleDataSet dataSetList = new DoubleDataSet("Measured Rate");

		chart.getDatasets().clear();

		// setto il dato
		for (int i = 0; i < rateData.size(); i++) {
			double x = rateData.get(i)[0];
			double rate = rateData.get(i)[1];
			dataSetList.add(x, rate);
		}

		chart.getDatasets().addAll(dataSetList);

	}

}
```

**Reply by wirew0rm** [reply]
The two most promising part to lock are:
a) the axis settings: have you tried to e.g. disable the logarithmic transformation?
b) updating of the data. Always completely replacing your dataset with a new one is pretty much the brute force way. You could always use the same dataset and just update the data. If you call plotData at a very high frequency, it might be that the dataset in the chart gets changed before the plugin has a chance to get the plugin data... If you only update every once in a while this should be fine, so it depends on your usecase.

Also: did you try to just copy an application from the sample repository to check if the plugin works in general?

**Reply by sangiorgi-inaf** [reply]
I call plot method just once, however I removed the clear dataset and log scale and doesn't work.
I've try the GaussianFitSample from the sample repository and add the plugin, doesn't work.

Maybe could be a problem with jdk 17 and/or javafx 18?

**Reply by wirew0rm** [reply]
Looks like there was a regression with this plugin, i can reproduce that it does not work for most samples while it works in the editDataSetSample (only one of the 2 plots) and the unit test. As I cannot make any promises as to when i get to fix it, i'd suggest to look at that sample and try to reproduce from there.

**Comment by sangiorgi-inaf** [comment]
thanks for your answers, i will try with parameter.

In a previous implementation of the same code, but with the old 8.1.5 library, I've used the tooltip without parameter and it worked fine.
